### PR TITLE
[WIP] Upgrade SQLAlchemy

### PR DESCRIPTION
**Pull request by tmpayton**
## Summary (required)

- Resolves #issue_number

(Include a summary of proposed changes and connect issue below)

### Required reviewers

(Include who is required to review prior to merge. For example: One designer and two front end developer reviews are required prior to merge)

## Impacted areas of the application

General components of the application that this PR will affect:

-  

## Screenshots

(Include a screenshot of the new/updated features in context (“in the wild”). If it is an interface change, include both before and after screenshots)

## Related PRs

Related PRs against other branches:

branch | PR
------ | ------
fix/other_pr | [link]()
feature/other_pr | [link]()

## How to test

(Include any information that may be helpful to the reviewer(s). This might include links to sample pages to test or any local environmental setup that is unusual such as environment variable (never credentials), API version to point to, etc)

- 

## System architecture updates (if applicable)

(If this pull request changes our [current system diagram](https://github.com/fecgov/FEC/wiki/2.-FEC-system-diagram), include a description of those changes here and create a new ticket to update the system diagram)

### CODE REVIEW

File: webservices/sorting.py [hunk 1]
```diff
@@ -77,8 +77,8 @@ def sort(query, key, model, aliases=None, join_columns=None, clear=False,
     # In this case, use the string name of the sort key.
     sort_model = (
         model
-        if len(query._entities) == 1 and hasattr(query._entities[0], 'mapper')
-        else None
+        # if len(query._entities) == 1 and hasattr(query._entities[0], 'mapper')
+        # else None
```
Comment (tmpayton) [inline webservices/sorting.py:81]: I still need to figure out a solution for this. Select does not contain the attribute _entities.

File: webservices/utils.py [hunk 2]
```diff
@@ -107,7 +108,7 @@ def fetch_page(
             index_column=index_column,
             nulls_last=nulls_last,
         )
-    paginator = paginators.OffsetPaginator(query, kwargs["per_page"], count=count)
+    paginator = paginators.OffsetPaginator(query, kwargs["per_page"], session, count=count)
```
Comment (tmpayton) [inline webservices/utils.py:111]: I'm going to look into binding the session so hopefully we won't have to pass it in!

File: webservices/common/counts.py
Comment (cnlucas) [inline webservices/common/counts.py:1]: This is very small and not helpful-- but I don't think you need to import sqlalchemy--both select and func are avail through the instance of sqlalchemy brought in through flask-sqlalchemy db

I think we import it quite a bit when we don't need to? Let me know if there's something I'm missing about that

File: webservices/sorting.py (hunk 1 above)
Comment (cnlucas) [reply webservices/sorting.py:81]: I think we have to use column_descriptions.

**Comment by cnlucas** [comment]
We are postponing this upgrade because it will require some major re-writing to how we are connecting to the database.

We discovered that we can no longer reset SQLALCHEMY_DATABASE_URI. Engines are now created when calling init_app rather than the first time they are accessed. The current recommended way of getting around this issue is to build an app factory. These changes would require us to re-create the app when we need to reset the configurations. This is extremely high risk for us.

Here's a helpful discussion of the issue:
https://github.com/pallets-eco/flask-sqlalchemy/issues/1172

We will be watching and continuing to work on solutions in case this upgrade becomes more urgent.

**Pull request closed without merging**